### PR TITLE
add a neon variant to libpng

### PR DESCRIPTION
**Pull request by hppritcha**
this turned out to be useful for building paraview for arm.

Signed-off-by: Howard Pritchard <howardp@lanl.gov>

### CODE REVIEW

File: var/spack/repos/builtin/packages/libpng/package.py [hunk 1]
```diff
@@ -35,14 +35,25 @@ class Libpng(AutotoolsPackage):
 
     depends_on('zlib@1.0.4:')  # 1.2.5 or later recommended
 
+    #
+    # Only set this if using a more recent GCC and building for ARM.
+    #
+    variant('arm-neon', default=False, description='Build ARM neon support')
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/libpng/package.py]: @hppritcha do you know what GCC version this requires? You could add a `conflicts` statement to make sure users don't try with a version of GCC that is too old.

(Ideally you could also do that to restrict the variant to ARM but that's difficult without support for syntax like `!platform=arm` which isn't supported right now)

File: var/spack/repos/builtin/packages/libpng/package.py [hunk 2]
```diff
@@ -35,14 +35,25 @@ class Libpng(AutotoolsPackage):
...
             #   https://sourceforge.net/p/libpng/bugs/210/#33f1
             # '--with-zlib=' + self.spec['zlib'].prefix,
             'CPPFLAGS={0}'.format(self.spec['zlib'].headers.include_flags),
-            'LDFLAGS={0}'.format(self.spec['zlib'].libs.search_flags)
+            'LDFLAGS={0}'.format(self.spec['zlib'].libs.search_flags),
         ]
+
+        if spec.satisfies('@1.6.34:'):
+            if '+arm-neon' in spec:
+                args.append('--enable-arm-neon=on')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libpng/package.py]: Probably want to explicitly disable this in an else clause

Comment (hppritcha) [reply var/spack/repos/builtin/packages/libpng/package.py]: added, thanks for pointing out

**Comment by adamjstewart** [comment]
@alalazo Is this PR still needed with the new microarchitecture support in Spack?

**Comment by chuckatkins** [comment]
I'd rather see the variant dropped entirely and just make it conditional on the presence of neon:
```python
if spec.satisfies('@1.6.34:'):
    args.append('--enable-arm-neon=' + ('on' if 'neon' in spec.target else 'off'))
```

**Comment by chuckatkins** [comment]
So, digging more into the `configure.ac`, I think this can be changed entirely to be non-arch specific and just pass `--enable-hardware-optimizations` always if the requisite version is met.

**Comment by alalazo** [comment]
@hppritcha Can you check https://github.com/spack/spack/pull/12582#issuecomment-561785668 and other comments and see if this PR is worth updating or dropping?

**Comment by adamjstewart** [comment]
No response from @hppritcha so I'm assuming this PR is no longer necessary

**Pull request closed without merging**